configuration/passport.js

```diff
@@ -0,0 +1,38 @@
+const passport = require( 'passport' )
+const LocalStrategy = require( 'passport-local' ).Strategy
+
+const User = require( '../database/user' )
+
+const strategy = new LocalStrategy({
+    usernameField: 'email',
+    session: false,
+    passReqToCallback: true
+}, ( request, email, password, done ) => {
+  User.findByEmail( email )
+    .then( user => {
+      if( User.validPassword( password,  user.password ) ) {
+        return done( null, user )
+      } else {
+        return done( null, false, request.flash( 'message', 'Oops! Wrong password.' ))
+      }
+    })
+    .catch( error => {
+      return done( null, false, request.flash( 'message', 'Seems we don\'t have your email...' ) )
+    })
+})
+
+passport.serializeUser( ( user, done ) => done( null, user.id ) )
+
+passport.deserializeUser( ( id, done ) => {
+  User.findById( id )
+    .then( user => {
+      return done( null, user )
+    })
+    .catch ( error => {
+      console.log( error )
```

Comment:
For logging errors, you can use the `console.error` method. https://nodejs.org/api/console.html#console_console_error_data_args